Inputs/CurrencyInput/functions/useCurrencyInputLayout.ts

```diff
@@ -0,0 +1,150 @@
+import type { Ref } from 'vue'
+import type { ICurrencyInputProps } from '~/components/Inputs/CurrencyInput/types/currency-input-props.type'
+import type { ICurrencyOptions } from '~/components/Inputs/CurrencyInput/types/currency-options.type'
+
+export function useCurrencyInputLayout(
+  props: ICurrencyInputProps,
+  emits: (e: 'update:modelValue', val?: number | null | undefined) => void,
+  input: Ref<HTMLInputElement | undefined>
+) {
+  // Layout
+  const inputValue = ref('0,00')
+  const unmaskedInputValue = computed<number>(() => unmasking(inputValue.value))
+  const prevValue = ref(props.min?.toString() ?? 0)
+
+  const currencyOptions = ref<ICurrencyOptions>()
+
+  // Functions
+  function init() {
+    currencyOptions.value = {
```

Comment:
why not put this above, when creating the variable, and then just use something like `defu` to merge the object? Now we need to use `!` all over the place